client/api/notebook.py

```diff
@@ -21,6 +23,8 @@ def run(self, protocol, messages, **kwargs):
         return self.assignment.protocol_map[protocol].run(messages, **kwargs)
 
     def auth(self, force=False, inline=True):
+        if not force and login_with_env(self.assignment):
```

Comment:
Yes - the Data Science packages use the notebook api for interacting with OK. Checkout a data8 assignment to see how it works.